test/compilable/traits.d

```diff
@@ -86,3 +86,50 @@ struct Outer
 static assert(__traits(getLocation, Outer.Nested)[1] == 82);
 static assert(__traits(getLocation, Outer.method)[1] == 84);
 
+/******************************************/
+// https://issues.dlang.org/show_bug.cgi?id=19902
+// Define hasElaborateCopyConstructor trait
+
+struct S
+{
+    this (ref S rhs) {}
+}
+
+struct OuterS
+{
+    struct S
+    {
+        this (ref S rhs) {}
+    }
+}
+
+void foo(T)()
+{
+    struct S(U)
+    {
+        this (ref S rhs) {}
+    }
+    static assert (__traits(hasElaborateCopyConstructor, S!int));
+}
+
+struct U(T)
+{
+    this (ref U rhs) {}
+}
+
+struct SPostblit
+{
+    this(this) {}
+}
+
+struct NoCpCtor { }
+
+static assert(__traits(hasElaborateCopyConstructor, S));
+static assert(__traits(hasElaborateCopyConstructor, OuterS.S));
+static assert(__traits(compiles, foo!int));
+static assert(__traits(compiles, foo!S));
+static assert(__traits(hasElaborateCopyConstructor, U!int));
+static assert(__traits(hasElaborateCopyConstructor, U!S));
+static assert(__traits(hasElaborateCopyConstructor, SPostblit));
```

Comment:
Can you explain why a library solution would return true here?

Reply:
This is what the library implementation of `hasElaborateCopyConstructor` [does](https://github.com/dlang/druntime/pull/2709/files#diff-f1322962163674c1ee8c7a018193ab45L261)